src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1113,12 +1113,6 @@ void java_lang_Class::create_mirror(Klass* k, Handle class_loader,
   assert(k != nullptr, "Use create_basic_type_mirror for primitive types");
   assert(k->java_mirror() == nullptr, "should only assign mirror once");
 
-  // Use this moment of initialization to cache modifier_flags also,
-  // to support Class.getModifiers().  Instance classes recalculate
-  // the cached flags after the class file is parsed, but before the
-  // class is put into the system dictionary.
```

Comment:
Is this comment already out-of-date? I'm trying to see where the flags are recomputed after parsing.

Reply:
And they're not the same as access flags.

Reply:
The "recalculate" is what is throwing me - that suggests the flags were already "calculated" somewhere. Does it really mean just "calculate" and that we do it between parsing and adding to SD? (Not that I know what the significance of that placement is meant to be.)

Reply:
So I did some old code reading.  We used to create the mirror when creating InstanceKlass, then later in ClassFileParser, we recalculated the modifier flags after the inner class attributes are added to the InstanceKlass.  Now we create the mirror after all of this.  Adding to the SystemDictionary is the point where the class is published so they have to be correct for that, which is what that part of the comment meant.